File: src/main/java/io/github/takenoha/towerdefense/paper/RaidSealItemIdentity.java
package io.github.takenoha.towerdefense.paper;

import java.util.Objects;
import java.util.UUID;

/** The stable identity carried by one physical "襲撃の印" item. */
public record RaidSealItemIdentity(UUID sealId, long stageLevel) {
    public RaidSealItemIdentity {
        Objects.requireNonNull(sealId, "sealId");
        if (stageLevel <= 0L) {
            throw new IllegalArgumentException("stageLevel must be positive");
        }
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/RaidSealItemIdentity.kt
package io.github.takenoha.towerdefense.paper

import java.util.Objects
import java.util.UUID
import kotlin.jvm.JvmRecord

/** The stable identity carried by one physical "襲撃の印" item. */
@JvmRecord
data class RaidSealItemIdentity(
    val sealId: UUID,
    val stageLevel: Long,
) {
    init {
        Objects.requireNonNull(sealId, "sealId")
        require(stageLevel > 0L) { "stageLevel must be positive" }
    }
}
